modules/core/src/main/scala/org/tessellation/domain/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -16,3 +16,13 @@ trait GlobalSnapshotStorage[F[_]] {
   )(address: Address): F[Option[StateChannelSnapshotBinary]]
 
 }
+
+trait GlobalSnapshotStorageV2[F[_]] {
+
+  def prepend(snapshot: GlobalSnapshot): F[Unit]
```

Comment:
I think that `prepend` should return `F[Boolean]` so that we can tell if the new snapshot passed a validation against the current head.

Reply:
If we're going to check against Next/Prev, then yes

Reply:
I think we want to have this kind of validation.